go.mod

```diff
@@ -18,13 +18,17 @@
 module github.com/gocql/gocql
 
 require (
-	github.com/bitly/go-hostpool v0.0.0-20171023180738-a3a6125de932 // indirect
-	github.com/bmizerany/assert v0.0.0-20160611221934-b7ed37b82869 // indirect
 	github.com/golang/snappy v0.0.3
 	github.com/hailocab/go-hostpool v0.0.0-20160125115350-e80d13ce29ed
-	github.com/kr/pretty v0.1.0 // indirect
-	github.com/stretchr/testify v1.3.0 // indirect
+	github.com/pierrec/lz4/v4 v4.1.8
```

Comment:
As I see from lz4 [repo](https://github.com/pierrec/lz4/blob/v4.1.8/go.mod) for release 4.1.8 go version in `go.mod` is set to `1.14`, but in the gocql `go.mod` go version is set to `1.13`.

Also, this PR bumps version of `github.com/stretchr/testify` from 1.7.0 to [1.9.0](https://github.com/stretchr/testify/blob/v1.9.0/go.mod) that requires go of version at least 1.17, The [1.7.0](https://github.com/stretchr/testify/blob/v1.7.0/go.mod) version only needs go 1.13.

@joao-r-reis should this PR change the go version here due to removing lz4 submodule?

Reply:
Hmm I don't think it matters if we change the version on go.mod here or not, the current version in go.mod is already incorrect and has been for a while anyway so I think we can just keep it unchanged in this PR and make the change on the v5 support PR to 1.19